xcodeproj/internal/library_targets.bzl

```diff
@@ -54,10 +54,11 @@ def process_library_target(
     build_settings = {}
 
     product_name = ctx.rule.attr.name
+    product_module_name = get_product_module_name(ctx = ctx, target = target)
```

Comment:
This will call `derive_module_name()` for Swift targets that don't set a custom module name. We don't want to send that to the generator.

How about we change `get_product_module_name` to return two values: the actual attribute, and the build setting value? We would set the build setting value like we have now, but we would use the actual attribute in `ConsolidatedTarget`.

Reply:
I thought about creating a separate method to retrieve only the value of an attribute, but that would require executing `getattr` twice. I assume that this method has a performance cost, so I decided to follow your suggestion.